ext/drreg/drreg.c

```diff
@@ -65,17 +65,49 @@
 
 #define PRE instrlist_meta_preinsert
 
-/* This should be pretty hard to exceed as there aren't this many GPRs */
+/* We add additional spill slots to cater for client spills.
+ * The number of spills should be pretty hard to exceed as there aren't
+ * that many GPRs.
+ */
 #define MAX_SPILLS (SPILL_SLOT_MAX + 8)
+/* We choose the number of available slots for spilling simds to match their
+ * theoretical max number for a given build.
+ */
+#define MAX_SIMD_SPILLS DR_NUM_SIMD_VECTOR_REGS
 
-#define AFLAGS_SLOT 0 /* always */
+#ifdef X86
+#    define XMM_REG_SIZE 16
+#    define YMM_REG_SIZE 32
+#    define ZMM_REG_SIZE 64
+#    define SIMD_REG_SIZE ZMM_REG_SIZE
+#else
+/* FIXME i#3844: NYI on ARM */
+#endif
 
-/* We support using GPR registers only: [DR_REG_START_GPR..DR_REG_STOP_GPR] */
+#define AFLAGS_SLOT 0 /* always */
 
+/* Liveliness states for gprs */
```

Comment:
I actually wanted this change this and thought since it is already used I don't mention it. Good to see you want to change it as well. Let's do it, done.

Reply:
Already used where?  "liveliness" is not used anywhere.